docs/src/modules/components/ApiPage.js

```diff
@@ -302,7 +302,14 @@ export default function ApiPage(props) {
   if (source === '@mui/x-date-pickers' || source === '@mui/x-date-pickers-pro') {
     packages.forEach((pkg) => {
       // e.g. import { DatePicker } from '@mui/x-date-pickers/DatePicker';
-      imports.push(`import { ${pkg.componentName} } from '${pkg.packageName}/${componentName}';`);
+      if (
+        packages.length === 1 ||
+        pkg.packageName === '@mui/x-date-pickers-pro' ||
+        pkg.componentName.startsWith('Adapter') ||
+        pkg.componentName === 'LocalizationProvider'
+      ) {
+        imports.push(`import { ${pkg.componentName} } from '${pkg.packageName}/${componentName}';`);
+      }
```

Comment:
I have done a couple of iterations to be able to use the same logic with the core components https://github.com/mui/material-ui/blob/master/docs/src/modules/components/ApiPage.js, as it seems that this component could be used purely for solving rendering problems. I don't see why there should be any differences.

What do you think about we move this logic outside of the ApiPage rendering logic?

Reply:
The second benefit, with the output inside the .json file, we could more easily debug the output. It seems that this logic is wrong here: 

<img width="778" alt="Screenshot 2023-05-13 at 13 16 14" src="https://github.com/mui/mui-x/assets/3165635/4af4a19d-7ba1-43e0-9dfd-2e4af01da6ca">

https://deploy-preview-8936--material-ui-x.netlify.app/x/api/date-pickers/mobile-time-picker/#import

Crash https://codesandbox.io/s/happy-wu-b7q2mc?file=/demo.tsx

```jsx
import { MobileDatePicker } from "@mui/x-date-pickers-pro/MobileDatePicker";
```

Reply:
> It seems that this logic is wrong here

Yes, the `MobileDatePicker` should be in the community instead of the pro

Reply:
Also in favor of improving the `json` 
I'm already listing the packages, we could also add a boolean for nested import.